### ISSUE TITLE
False positives on background-only images during validation (YOLO11 Segmentation)

### ISSUE THREAD

**Issue by kkk2705** [ultralytics/ultralytics#20333]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hi, I'm training a YOLO11 segmentation model to detect mature and immature oranges. I added around 170 background-only images (with no fruits) to help reduce false detections. Each image has an empty .txt label file, and everything is set correctly in data.yaml.

Still, during validation, the model sometimes detects reddish leaves as mature fruits. I’ve attached an example where a leaf-only region was predicted as fruit.

I just want to confirm:
Are background images with empty label files used during training in YOLOv8 segmentation? Or should I handle them differently (e.g., label as a “background” class)? Any advice to reduce false positives like this would help a lot. Thanks!

![Image](https://github.com/user-attachments/assets/1ed80d04-1753-4787-86dd-760f07b747e7)

![Image](https://github.com/user-attachments/assets/81f61c32-a578-4cda-bc1a-0219d55b2ff7)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @kkk2705, thank you for reaching out and for providing detailed context and example images! 🚀 This is an automated response to help you get the most relevant support. An Ultralytics engineer will review your question and assist you soon.

We recommend reviewing the [Docs](https://docs.ultralytics.com/) for guidance on segmentation tasks, as well as best practices for [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage. Your question about background-only images is a common one, and you may find helpful information in the [FAQs](https://docs.ultralytics.com/help/) and [model training tips](https://docs.ultralytics.com/guides/model-training-tips/).

If this is a 🐛 Bug Report, please provide a [minimum reproducible example (MRE)](https://docs.ultralytics.com/help/minimum-reproducible-example/) if you haven’t already. This greatly helps us investigate and resolve the issue faster.

For training or validation questions, please include as much information as possible, such as:
- Example images (which you’ve provided—thanks!)
- Training logs or relevant code snippets
- Details from your `data.yaml` and training configuration

You can join the Ultralytics community for faster interactions and more support:
- [Discord](https://discord.com/invite/ultralytics) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com/) for in-depth discussions
- [Subreddit](https://reddit.com/r/Ultralytics) to share and learn with others

## Upgrade

To ensure you’re running the latest features and fixes for YOLO11 segmentation, please upgrade the `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run in any of these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you for helping us improve Ultralytics! An engineer will follow up to assist you further.

**Comment by Y-T-G** [comment]
Did it show the background count jn training logs?

**Comment by kkk2705** [comment]
Yes in confusion matrix it showed count it predicted ad false postive

![Image](https://github.com/user-attachments/assets/edc5019a-6a7f-41a9-a09d-0bd174612311)

. The background images were added by cropping same image, but even the issue persists.

**Comment by Y-T-G** [comment]
I mean when the training starts, the logs show the count of images including the count of background. Can you check that?

**Comment by Y-T-G** [comment]
Did you train with the background images?

**Comment by kkk2705** [comment]
Yes I trained with background images and even it showed while upload in training also
train/labels... 2392 images, 870 backgrounds, 0 corrupt

**Comment by kkk2705** [comment]
Yeah, I tried data augmentation but others I will try. I have no idea of (4) regarding advanced techniques like focal losses. could you please give any source?

**Comment by glenn-jocher** [comment]
Focal loss helps models focus on hard-to-classify examples by reducing the loss contribution from easy-to-classify examples. It's particularly useful for addressing class imbalance issues like yours.

YOLO11 already implements a version of focal loss by default, but you can adjust it through the `hyp.yaml` file or using the `fl_gamma` parameter during training. Higher gamma values (like 1.5-2.0) make the model focus more on difficult examples.

For resources on focal loss:
1. The original paper: [Focal Loss for Dense Object Detection](https://arxiv.org/abs/1708.02002)
2. Our implementation details in the [Ultralytics docs](https://docs.ultralytics.com/guides/yolo-common-issues/)

You might also try gradually increasing your `conf` threshold during validation to find the optimal balance between reducing false positives while maintaining true positive detections.
